sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/RowConverter.scala

```diff
@@ -3,246 +3,224 @@
-      case LongType => objectMapper.convertValue(element.asInstanceOf[Long], classOf[JsonNode])
-      case FloatType => objectMapper.convertValue(element.asInstanceOf[Float], classOf[JsonNode])
-      case NullType => objectMapper.nullNode()
-      case DecimalType() => if (element.isInstanceOf[Decimal]) {
-        objectMapper.convertValue(element.asInstanceOf[Decimal].toJavaBigDecimal, classOf[JsonNode])
-      } else if (element.isInstanceOf[java.lang.Long]) {
-        objectMapper.convertValue(new java.math.BigDecimal(element.asInstanceOf[java.lang.Long]), classOf[JsonNode])
-      } else {
-        objectMapper.convertValue(element.asInstanceOf[java.math.BigDecimal], classOf[JsonNode])
-      }
-      case StringType =>
-        if (isInternalRow) {
-          objectMapper.convertValue(element.asInstanceOf[UTF8String].toString, classOf[JsonNode])
-        } else {
-          objectMapper.convertValue(element.asInstanceOf[String], classOf[JsonNode])
-        }
-      case TimestampType => if (element.isInstanceOf[java.lang.Long]) {
-        objectMapper.convertValue(element.asInstanceOf[java.lang.Long], classOf[JsonNode])
-      } else {
-        objectMapper.convertValue(element.asInstanceOf[Timestamp].getTime, classOf[JsonNode])
-      }
-      case arrayType: ArrayType => arrayTypeRouterToJsonArray(arrayType.elementType, element, isInternalRow)
-
-      case mapType: MapType if isInternalRow =>
-        mapType.keyType match {
-          case StringType =>
-            // convert from UnsafeMapData to scala Map
-            val unsafeMap = element.asInstanceOf[UnsafeMapData]
-            val keys: Array[String] = unsafeMap.keyArray().toArray[UTF8String](StringType).map(_.toString)
-            val values: Array[AnyRef] = unsafeMap.valueArray().toObjectArray(mapType.valueType)
-            mapTypeToObjectNode(mapType.valueType, keys.zip(values).toMap, isInternalRow)
-          case _ => throw new Exception(
-            s"Cannot cast $element into a Json value. MapTypes must have keys of StringType for conversion into a Document"
-          )
+    }
+
+    // scalastyle:off
+    private def convertSparkDataTypeToJsonNode(fieldType: DataType, rowData: Any) : JsonNode = {
+        fieldType match {
+            case StringType if rowData.isInstanceOf[String] => objectMapper.convertValue(rowData.asInstanceOf[String], classOf[JsonNode])
+            case StringType if rowData.isInstanceOf[UTF8String] => objectMapper.convertValue(rowData.asInstanceOf[UTF8String].toString, classOf[JsonNode])
+            case BinaryType => objectMapper.convertValue(rowData.asInstanceOf[Array[Byte]], classOf[JsonNode])
+            case BooleanType => objectMapper.convertValue(rowData.asInstanceOf[Boolean], classOf[JsonNode])
+            case DoubleType => objectMapper.convertValue(rowData.asInstanceOf[Double], classOf[JsonNode])
+            case IntegerType => objectMapper.convertValue(rowData.asInstanceOf[Int], classOf[JsonNode])
```

Comment:
These conversions have been part of heated discussion sometimes. Common challenges are:

Long, Double, Float, Decimal --> When converting them to json as number you always risk loosing precision - so customers sometimes need to convert it to string instead to preserve precisions - this of course makes it impossible to use any aggregates/numeric functions within Cosmos native queries.

DateType/TimestampType --? Contention whether to convert into Long (Epoch of seconds vs. Epoch of milliseconds) or into a string representation - like ISO 8601-2019

I am wondering whether we should make this configurable - and if so rather as a way of setting a global flag (preferTextualJsonRepresentation vs. prefernativeJsonRepresenttation which would be used for all applicable columns/data types - or whether this would need ot be configurable ona  per column level.
Ultimately tehre is always the workaround for custoemrs to do the conversion on the dta frame upstream/downstream - but I think this has been a contention point often enough to have a brief discussion to decide how we want to proceed. Makes sense?

Reply:
I added Json > Spark Row support for fields that are strings into DateType.Timestamp and DataType.Date by trying to parse the string as ISO 8601 with `+HH:mm` timezone notation or `Z` for UTC.

The Spark Row > Json serialization, that is something that might require extra configuration work and outside of the scope of this PR.